dulwich/refs.py

```diff
@@ -403,8 +403,9 @@ def get_peeled(self, name):
 class DiskRefsContainer(RefsContainer):
     """Refs container that reads refs from disk."""
 
-    def __init__(self, path):
+    def __init__(self, path, **kwargs):
         self.path = path
+        self.worktree_path = kwargs["worktree"]
```

Comment:
I think we should actually not make 'HEAD' a part of RefsContainer if it's working tree specific. Can you add a # TODO comment mentioning that.